### ISSUE TITLE
Kernel logs times are off by a bit more than one day

### ISSUE THREAD

**Issue by JulienPalard** [systemd/systemd#33293]
### systemd version the issue has been seen with

256-rc4

### Used distribution

Debian Trixie

### Linux kernel version used

6.7.12-amd64

### CPU architectures issue was seen on

None

### Component

systemd-journald

### Expected behaviour you didn't see

I expected "monotonic" times while reading logs from `journalctl -f` (at least with -o short-monotonic).

### Unexpected behaviour you saw

I saw logs from the future here, like from tomorrow, a tad more than 24h. From 24h 36m and a few seconds in the future, or 88563 seconds but I bet this number is not a constant.

### Steps to reproduce the problem

I don't know yet, I only have a few hints.

### Additional program output to the terminal or log subsystem illustrating the issue

Boot happened with: `Jun 10 18:33:27 kalir systemd-journald[1318]: Journal started`

First occurence of time travel a few hours later, related to hibernation:

```text
Jun 10 21:57:50 kalir NetworkManager[1742]: <info>  [1718049470.7180] dhcp4 (wlp1s0): state changed no lease
Jun 11 00:12:24 kalir kernel: Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
Jun 10 21:57:50 kalir dbus-daemon[1682]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' requested by ':1.14' (uid=0 pid=1742 comm="/usr/sbin/NetworkManager --no-daemon")
```

Now, at the time of writing, we're at `Jun 12 13:39:15`, but I already see events from the future by searching `Jun 13` in the logs:

```
Jun 12 13:29:15 kalir rtkit-daemon[2213]: Supervising 0 threads of 0 processes of 0 users.
Jun 12 13:29:15 kalir rtkit-daemon[2213]: Supervising 0 threads of 0 processes of 0 users.
Jun 12 13:29:31 kalir gdm-password][104683]: gkr-pam: unlocked login keyring
Jun 12 13:29:32 kalir NetworkManager[1742]: <info>  [1718191772.0770] agent-manager: agent[4117da6764403ca4,:1.78/org.gnome.Shell.NetworkAgent/1000]: a
gent registered
Jun 13 15:05:49 kalir kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=14
Jun 13 15:05:49 kalir kernel: [drm:amdgpu_mes_reg_write_reg_wait [amdgpu]] *ERROR* failed to reg_write_reg_wait
Jun 13 15:05:49 kalir kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=14
Jun 13 15:05:49 kalir kernel: [drm:amdgpu_mes_reg_write_reg_wait [amdgpu]] *ERROR* failed to reg_write_reg_wait
Jun 12 13:29:50 kalir rtkit-daemon[2213]: Supervising 0 threads of 0 processes of 0 users.
Jun 12 13:29:50 kalir rtkit-daemon[2213]: Supervising 0 threads of 0 processes of 0 users.
```

I notice that the `kernel` lines are in the future while the **not** kernel ones are good.

Running journalctl with `-o short-monotonic` does not help, I see the same jumps:

```
[ 4258.534832] kalir dbus-daemon[1682]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dis
patcher.service' requested by ':1.14' (uid=0 pid=1742 comm="/usr/sbin/NetworkManager --no-daemon")
[ 4258.535896] kalir avahi-daemon[1679]: Withdrawing address record for 192.168.1.34 on wlp1s0.
[12332.553347] kalir kernel: Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[ 4258.535939] kalir avahi-daemon[1679]: Leaving mDNS multicast group on interface wlp1s0.IPv4 with address 192.168.1.34.
[12332.557761] kalir kernel: Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[12332.561601] kalir kernel: Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[12332.564929] kalir kernel: Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[ 4258.548293] kalir avahi-daemon[1679]: Interface wlp1s0.IPv4 no longer relevant for mDNS.
```

**Comment by JulienPalard** [comment]
[Added a missing amdgpu module](https://community.frame.work/t/error-mes-failed-to-response-msg-14/52719/2) causing many log messages, and I haven't seen the time wrap reproduce since (only tested for one hour...).

**Comment by yuwata** [comment]
Could you provide the relevant lines with `journalctl -o export` ?

**Comment by yuwata** [comment]
Is this new in v256?

**Comment by JulienPalard** [comment]
> Could you provide the relevant lines with `journalctl -o export` ?

I'm using volatile journals, and had to reboot to load the missing firmware, so ... next time :)

> Is this new in v256?

No idea, I only have this laptop for a few days... I have no history with it.

**Comment by JulienPalard** [comment]
OK I've seen it again in the logs:

```
Jun 13 08:43:48 kalir NetworkManager[1356]: <info>  [1718261028.0576] dhcp4 (enxc03eba12b054): activation: beginning transaction (timeout in 45 seconds)
Jun 13 22:29:47 kalir kernel: r8152 10-2.4:1.0 enxc03eba12b054: carrier on
Jun 13 08:43:48 kalir avahi-daemon[1287]: Joining mDNS multicast group on interface enxc03eba12b054.IPv6 with address fe80::b5f9:b6b5:6c:4d42.
```

so it was not caused by the missing amdgpu firmware. The missing firmware caused **me** to read the logs and spot the time shift.

Ohhhh this is interesting, it looks like the time is shifting slowly:

First a one second shift (kernel is in the future):
```
Jun 12 16:47:19 kalir NetworkManager[1356]: <info>  [1718203639.9930] dhcp4 (wlp1s0): state changed no lease
Jun 12 16:47:20 kalir kernel: Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
Jun 12 16:47:20 kalir kernel: Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
Jun 12 16:47:19 kalir dbus-daemon[1290]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' req>
```

Later there's a few minute of shift:

```
Jun 12 16:55:12 kalir wpa_supplicant[1357]: wlp1s0: CTRL-EVENT-SUBNET-STATUS-UPDATE status=0
Jun 12 17:02:57 kalir kernel: wlp1s0: associated
Jun 12 16:55:12 kalir NetworkManager[1356]: <info>  [1718204112.1294] device (wlp1s0): supplicant interface state: associating -> 4way_handshake
```

and more:

```
Jun 12 17:15:32 kalir systemd-sleep[32566]: System returned from sleep operation 'suspend'.
Jun 12 17:27:59 kalir kernel: PM: suspend exit
Jun 12 17:15:32 kalir systemd[1]: Starting fwupd-refresh.service - Refresh fwupd metadata and update motd...
```

and a few hours:

```
Jun 13 08:43:45 kalir mtp-probe[42932]: bus: 10, device: 4 was not an MTP device
Jun 13 22:29:44 kalir kernel: r8152 10-2.4:1.0 enxc03eba12b054: renamed from eth0
Jun 13 08:43:45 kalir NetworkManager[1356]: <info>  [1718261025.2358] device (eth0): interface index 3 renamed iface from 'eth0' to 'enxc03eba12b054'
```

I confirm that the kernel is in the future, as as I write this we're Jun 13 8:54 and kernel logs are from Jun 13 22:29.

It looks related to suspend: the longer my laptop spends time in suspend, the more the shift grows.

**Comment by JulienPalard** [comment]
OK, a "-o export" of following lines:

```
Jun 13 20:58:42 kalir wpa_supplicant[1158]: wlp1s0: CTRL-EVENT-SUBNET-STATUS-UPDATE status=0
Jun 13 23:41:52 kalir kernel: wlp1s0: associated
Jun 13 20:58:42 kalir NetworkManager[1157]: <info>  [1718305122.3335] device (wlp1s0): supplicant interface state: associating -> 4way_handshake
```

```
__CURSOR=s=b6c7d9a1c30a4ef993464c9bb94617b3;i=1193;b=53a9307a18b44568aa27ac97617bba2e;m=e836fb1e;t=61aca166e9d8d;x=38598ec036e2f408
__REALTIME_TIMESTAMP=1718305122327949
__MONOTONIC_TIMESTAMP=3895917342
__SEQNUM=4499
__SEQNUM_ID=b6c7d9a1c30a4ef993464c9bb94617b3
_BOOT_ID=53a9307a18b44568aa27ac97617bba2e
PRIORITY=5
_MACHINE_ID=fc0069bdd11a4fc791a40225ebfdaacb
_HOSTNAME=kalir
_RUNTIME_SCOPE=system
SYSLOG_FACILITY=3
_UID=0
_SELINUX_CONTEXT
^K^@^@^@^@^@^@^@unconfined

_SYSTEMD_SLICE=system.slice
_CAP_EFFECTIVE=1ffffffffff
_TRANSPORT=syslog
SYSLOG_IDENTIFIER=wpa_supplicant
SYSLOG_PID=1158
_PID=1158
_GID=101
_COMM=wpa_supplicant
_EXE=/usr/sbin/wpa_supplicant
_CMDLINE=/usr/sbin/wpa_supplicant -u -s -O "DIR=/run/wpa_supplicant GROUP=netdev"
_SYSTEMD_CGROUP=/system.slice/wpa_supplicant.service
_SYSTEMD_UNIT=wpa_supplicant.service
_SYSTEMD_INVOCATION_ID=d10dece9eac3431091a7200bc6acc822
MESSAGE=wlp1s0: CTRL-EVENT-SUBNET-STATUS-UPDATE status=0
SYSLOG_TIMESTAMP=Jun 13 20:58:42 
_SOURCE_REALTIME_TIMESTAMP=1718305122327904

__CURSOR=s=b6c7d9a1c30a4ef993464c9bb94617b3;i=1194;b=53a9307a18b44568aa27ac97617bba2e;m=e8370611;t=61aca166ea880;x=c6d4213a4e3c1742
__REALTIME_TIMESTAMP=1718305122330752
__MONOTONIC_TIMESTAMP=3895920145
__SEQNUM=4500
__SEQNUM_ID=b6c7d9a1c30a4ef993464c9bb94617b3
_BOOT_ID=53a9307a18b44568aa27ac97617bba2e
_TRANSPORT=kernel
SYSLOG_FACILITY=0
SYSLOG_IDENTIFIER=kernel
_MACHINE_ID=fc0069bdd11a4fc791a40225ebfdaacb
_HOSTNAME=kalir
_RUNTIME_SCOPE=system
PRIORITY=6
MESSAGE=wlp1s0: associated
_SOURCE_MONOTONIC_TIMESTAMP=13685686976

__CURSOR=s=b6c7d9a1c30a4ef993464c9bb94617b3;i=1195;b=53a9307a18b44568aa27ac97617bba2e;m=e8371118;t=61aca166eb386;x=a9aa585e374d16fc
__REALTIME_TIMESTAMP=1718305122333574
__MONOTONIC_TIMESTAMP=3895922968
__SEQNUM=4501
__SEQNUM_ID=b6c7d9a1c30a4ef993464c9bb94617b3
_BOOT_ID=53a9307a18b44568aa27ac97617bba2e
_MACHINE_ID=fc0069bdd11a4fc791a40225ebfdaacb
_HOSTNAME=kalir
_RUNTIME_SCOPE=system
PRIORITY=6
SYSLOG_FACILITY=3
_UID=0
_GID=0
_SELINUX_CONTEXT
^K^@^@^@^@^@^@^@unconfined

_SYSTEMD_SLICE=system.slice
_TRANSPORT=journal
SYSLOG_IDENTIFIER=NetworkManager
SYSLOG_PID=1157
NM_LOG_LEVEL=INFO
_PID=1157
_COMM=NetworkManager
_EXE=/usr/sbin/NetworkManager
_CMDLINE=/usr/sbin/NetworkManager --no-daemon
_CAP_EFFECTIVE=200534e2
_SYSTEMD_CGROUP=/system.slice/NetworkManager.service
_SYSTEMD_UNIT=NetworkManager.service
_SYSTEMD_INVOCATION_ID=8d23701e693141c895d3460c462ddabe
CODE_FILE=../src/core/devices/wifi/nm-device-wifi.c
NM_DEVICE=wlp1s0
NM_LOG_DOMAINS=WIFI,DEVICE
CODE_LINE=2467
MESSAGE=<info>  [1718305122.3335] device (wlp1s0): supplicant interface state: associating -> 4way_handshake
TIMESTAMP_MONOTONIC=13667.570865
TIMESTAMP_BOOTTIME=13684.570865
_SOURCE_REALTIME_TIMESTAMP=1718305122333549
```

**Comment by yuwata** [comment]
Thank you for the report!

**Comment by yuwata** [comment]
Fix is waiting in #33386.
